sdk/js/executor.js

```
@@ -0,0 +1,300 @@
+                if (err2 != null) {
+                    return [null, null, err2]
+                }
+                chainIDs.push(child.chainID)
+                cbcAddrs.push(childSegEvent.address)
+                eventFuncSigs.push(segmentFuncSig)
+                eventDatas.push(childSegEvent.raw.data)
+                eventSigs.push(childSegSig)
+            }
+        }
+        // Either this node is a leaf node or this node is an intermediate node and we have already collected all segment events for its child nodes.
+        // Get chain ap.
+        var web3 = await this.cmgr.chainAP(node.chainID)
+        // Get GPACT contract
+        var gpactAddr = this.gpacts.get(node.chainID)
+        var gpact = new web3.eth.Contract(gpactABI, gpactAddr)
+        // //
+        // Try 5 times, 5 seconds apart.
+        for (let i = 0; i < 5; i++) {
+            try {
+                var gasPrice = await web3.eth.getGasPrice()
+                var res = await gpact.methods.segment(chainIDs, cbcAddrs, eventFuncSigs, eventDatas, eventSigs, callPath).send({
+                    from: this.account,
+                    gas: 600000,
+                    gasPrice: gasPrice,
+                })
+                console.log("segment succeed")
+                console.log(res)
+                segmentEvent = res.events.Segment   
+                // Try to get event sig
+                // Try 60 times, 2 seconds apart.
+                var eventID = getEventID(node.chainID, segmentEvent)
+                for (let j = 0; j < 60; j++) {
+                    var sig = await this.ms.getSignature(eventID)
+                    if (sig != null) {
+                        segmentEventSig = sig
+                        break
+                    }
+                    console.log("fail to get event sig, try again...")
+                    await new Promise(r => setTimeout(r, 2000))
+                }
```

If all retry attempts in contacting the message store fail here, then this block fails silently. Should the `err` variable be set to inform the caller? 

comment applies to `start()` and `root()` as well.